test/core_primitive_io_test.cpp

```diff
@@ -0,0 +1,246 @@
+
+#include <gtest/gtest.h>
+#include "hoNDImage.h"
+#include "hoMRImage.h"
+#include "hoNDArray_elemwise.h"
+#include "io/primitives.h"
+
+TEST(TypeTests, hoNDImageTest) {
+
+    Gadgetron::hoNDImage<double, 2> im2d(12, 3);
+
+    im2d.set_pixel_size(0, 1.2);
+    im2d.set_pixel_size(1, 3.2);
+
+    im2d.set_origin(0, 0.23);
+    im2d.set_origin(1, 12.45);
+
+    im2d.set_axis(0, 0, 0.8);
+    im2d.set_axis(0, 1, 0.2);
+
+    im2d.set_axis(1, 0, 0.1);
+    im2d.set_axis(1, 1, 0.9);
+
+    Gadgetron::fill(im2d, 3.42);
+
+    std::ostringstream ostr;
+    Gadgetron::Core::IO::write(ostr, im2d);
+
+    std::string content = ostr.str();
+
+    std::istringstream istr;
+    istr.str(content);
+
+    Gadgetron::hoNDImage<double, 2> im2d_readed;
```

Comment:
nit: variable name > `im2d_read`....propagate throughout.

Reply:
change to im_readed

Reply:
It is the `readed` part of it that seems wrong. You mean past tense of "read", which is well "read", right?

Reply:
Got it!